tunnel/intercept/svcpoll.go

```diff
@@ -105,47 +109,65 @@ func diffServices(edgeServices []edge.Service, knownServices map[string]edge.Ser
 	return
 }
 
-func updateServices(context ziti.Context, interceptor Interceptor, resolver dns.Resolver, added, removed, all map[string]edge.Service) {
+func updateServices(context ziti.Context, interceptor Interceptor, resolver dns.Resolver, added, removed, all map[string]*entities.Service) {
 	log := pfxlog.Logger()
 	for _, svc := range added {
 		if stringz.Contains(svc.Permissions, "Dial") {
+			clientConfig := &entities.ServiceConfig{}
+			if found, err := svc.GetConfigOfType(entities.ClientConfigV1, clientConfig); found && err == nil {
+				svc.ClientConfig = clientConfig
+			} else if !found {
+				pfxlog.Logger().Infof("no service config of type %v for service %v", entities.ClientConfigV1, svc.Name)
+			} else if err != nil {
+				pfxlog.Logger().WithError(err).Infof("error decoding service config of type %v for service %v", entities.ClientConfigV1, svc.Name)
+			}
 			log.Infof("starting tunnel for newly available service %s", svc.Name)
 			err := interceptor.Intercept(svc, resolver)
 			if err != nil {
 				log.Errorf("failed to intercept service: %v", err)
 			}
 		}
 		if stringz.Contains(svc.Permissions, "Bind") {
-			if dialAddr, ok := svc.Tags["tunneler.dial.addr"]; ok {
+			serverConfig := &entities.ServiceConfig{}
+			found, err := svc.GetConfigOfType(entities.ServerConfigV1, serverConfig)
+
+			if found && err == nil {
+				svc.ServerConfig = serverConfig
 				log.Infof("Hosting newly available service %s", svc.Name)
-				go host(context, svc, dialAddr)
-			} else {
-				log.Warnf("service %v is hostable but is missing a dial address. Add a 'tunneler.dial.addr' tag to the service to fix", svc.Name)
+				go host(context, svc)
+			} else if !found {
+				log.WithError(err).Warnf("service %v is hostable but no server config of type %v is available", svc.Name, entities.ServerConfigV1)
+			} else if err != nil {
+				log.WithError(err).Errorf("service %v is hostable but unable to decode server config of type %v", svc.Name, entities.ServerConfigV1)
 			}
 		}
 	}
 
```

Comment:
We don't know here if the intercept above actually worked. Should we do anything different if it failed (if, for example, the config was missing or invalid)?

Reply:
Good question. As it is right now the address of a service with client config that fails to intercept will be added to the "all" map. It isn't ideal but I think it's acceptable, since the only harm that will come from it is a doomed call to StopIntercepting.

Avoiding this would require keeping track of services that we are aware of but not intercepting (lest we attempt to set up the intercept on every poll cycle).